packages/suite/src/components/suite/modals/TransactionDetail/components/AmountDetails.tsx

```diff
@@ -57,14 +63,22 @@ export const AmountDetails = ({ tx, isTestnet }: AmountDetailsProps) => {
     const [showFiat, setShowFiat] = useState(false);
 
     const tokenTransfer = tx.tokens.length > 0 ? tx.tokens[0] : undefined;
-    const amount = tokenTransfer ? tokenTransfer.amount : tx.amount;
+    const txAmount = formatNetworkAmount(tx.amount, tx.symbol);
+    const amount = tokenTransfer
+        ? formatAmount(tokenTransfer.amount, tokenTransfer.decimals)
+        : txAmount;
     const assetSymbol = tokenTransfer ? tokenTransfer.symbol : tx.symbol;
 
     const network = getNetwork(tx.symbol);
     const hideTotalIORows =
         network?.networkType === 'ripple' || network?.networkType === 'ethereum'; // don't show for eth, xrp
 
     const showHistoricalRates = showFiat && !tokenTransfer;
+    const totalInput = formatNetworkAmount(tx.details.totalInput, tx.symbol);
```

Comment:
Have you considered actually changing `FormattedCryptoAmount` right away as we discussed to avoid all those conversions?

Reply:
No but I have considered to put it at the end of the PR description to the **followups** section so we could decide who and when should do it. :)

Reply:
Yes, the goal here is to enable further work on coinjoin, rather then dig even deeper into the rabbit hole of refactoring. But maybe if you create also a followup issue linked to this PR?

Reply:
https://github.com/trezor/trezor-suite/issues/6057